tools/testsys/src/aws_resources.rs

```diff
@@ -173,8 +206,12 @@ impl AwsK8s {
                 conflicts_with: None,
                 agent: Agent {
                     name: "eks-provider".to_string(),
-                    image: testsys_images.eks_resource.clone(),
-                    pull_secret: testsys_images.secret.clone(),
+                    image: testsys_images
+                        .eks_resource_agent_image
+                        .as_ref()
+                        .cloned()
+                        .unwrap(),
```

Comment:
I think you could add a comment here explaining why this `unwrap` is safe. It took me a while to understand that the resulting `testsys_images` object will have a default value for `eks_resource_agent_image` regardless of the configurations provided by `Test.toml`.

Please do the same for any `unwrap` you left around that is safe to keep.